alchemist-full/build.gradle.kts

```diff
@@ -33,23 +32,49 @@ plugins {
     alias(libs.plugins.shadowJar)
 }
 
-dependencies {
-    runtimeOnly(rootProject)
-    rootProject.subprojects.filterNot { it == project }.forEach {
-        if (it.isMultiplatform) {
-            runtimeOnly(project(path = ":${it.name}", configuration = "default"))
-        } else {
-            runtimeOnly(it)
+kotlin {
+    jvm {
+        withJava()
+    }
+    sourceSets {
+        val commonMain by getting {
+            dependencies {
+                runtimeOnly(rootProject)
+                rootProject.subprojects.filterNot { it == project }.forEach {
+                    runtimeOnly(it)
+                }
+            }
+        }
+        val commonTest by getting {
+            dependencies {
+                implementation(rootProject)
+                implementation(alchemist("physics"))
+            }
         }
     }
-    testImplementation(rootProject)
-    testImplementation(alchemist("physics"))
 }
 
+/**
+ * @return a [FileCollection] containing the classpath of the multiplatform JVM projects.
+ * If unspecified, the classpath is not correctly built when including common code from MP dependencies,
+ * resulting in runtime errors.
+ */
+private fun mpClasspath(): FileCollection = listOf(
```

Comment:
This breaks the configuration avoidance API... I think we need to do a bit of pair programming to see whether this can be solved

Reply:
We did it